### PR TITLE
support resume training

### PR DESCRIPTION
**Pull request by L1aoXingyu**
断点续训的流程如下：
1. 根据 last_checkpoint 获得已经训练的 iter 数，计算 consumed samples；
2. 根据 consumed samples，构建 dataloader；
3. 根据 dataloader 进行 scale hyperparamters；
4. 构建 model/optim/scheduler；
5. 读取 model/optim/scheduler state_dict；

image

### CODE REVIEW

File: libai/trainer/default.py [hunk 1]
```diff
@@ -212,11 +212,18 @@ def __init__(self, cfg):
...
-        #     cfg.dataloader.cousumed_samples = ...
+        self.start_iter = 0
+        if cfg.train.resume:
+            save_file = os.path.join(cfg.train.output_dir, "last_checkpoint")
+            try:
+                with open(save_file, "r") as f:
+                    last_saved = f.read().strip()
+                self.start_iter = int(last_saved.split("_")[-1]) + 1
+            except IOError:
+                # If file doesn't exist, maybe because it has just been deleted.
```
Comment (rentainhe) [inline libai/trainer/default.py:223]: 我觉得这个地方可以加一个`logging`输出注释的这些信息~

**Review by rentainhe — APPROVED** [review]

File: libai/trainer/default.py (hunk 1 above)
Comment (L1aoXingyu) [reply libai/trainer/default.py:223]: 下面载入模型的地方有这个提示

image
